### DISCUSSION TITLE
Does compass need calibration after sensor rotation ?

### DISCUSSION THREAD

**Discussion by mrechte** [iNavFlight/inav#8029] [Q&A] [answered]
Hi. I did a compass calibration while the sensor was on the same plane than the drone chassis. Now I tilted 30° the GPS/compass unit on the pitch axis (in order to compensate the drone attitude while flying). I used the https://kernel-machine.github.io/INavMagAlignHelper/ tool to take that rotation into consideration, and entered the given figures with the CLI.

Do I have to recalibrate the compass, or is it going to confuse the system if I do it again ?

Thanks

**Comment by JulioCesarMatias** [comment, marked as answer]
> Hi. I did a compass calibration while the sensor was on the same plane than the drone chassis. Now I tilted 30° the GPS/compass unit on the pitch axis (in order to compensate the drone attitude while flying). I used the https://kernel-machine.github.io/INavMagAlignHelper/ tool to take that rotation into consideration, and entered the given figures with the CLI.
> 
> Do I have to recalibrate the compass, or is it going to confuse the system if I do it again ?
> 
> Thanks

yes, recalibrate the compass
